### PR TITLE
DB backup

### PR DESCRIPTION
**Pull request by jdk24**
### Info

pg_dump creates A LOT of warnings on timescaleDB. This is expected. 

Current output dir:  /home/backups/database/postgresql/

Testing: Build and start stack, exec container, either set crontab to every minute or execute script directly.
Check if backup is created in target dir.

### Backup script

Clean up script as we will never use some of the methods provided

pg_dumpall would back up giant road database. Only back up sauber_data. 
So: Hijack schema-only backup workflow: Make backup only of databases in DB_BACKUP_LIST (Lines 76..78

Remove host flag from psql, pg_dump: We're only working from local anyway

Harcode full config file path for clarity

Add additional info to output: Target directory and time of backup

### crontab

Commit altered crontab file to repo

### config

Do not check for user postgres - redundant on Docker

Adjust databases-to-back-up list name (see above)

### CODE REVIEW

**Review by JakobMiksch — APPROVED** [review]
Looks alright for me :+1

**Review by chrismayer — COMMENTED** [review]
Code looks good to me, but I could not start the DB service in the stack locally:

image

Maybe not related to this PR, but would be good if someone of you (@jdk @JakobMiksch) could fire up the stack and check if it runs.

**Comment by jdk24** [comment]
@chrismayer  Could you have a look in one DB service logfile and see if the end says something like [this](https://github.com/meggsimum/sauber-sdi-docker/pull/132/files#diff-c272aa557400cbf25004c8035bd2d4aaddd4286ae997a072bd8d54f1a5009005R146-L161)? I'm gonna guess it's missing [this](https://github.com/meggsimum/sauber-sdi-docker/pull/132/files#diff-c272aa557400cbf25004c8035bd2d4aaddd4286ae997a072bd8d54f1a5009005R1). Courtesy of not deleting my DB volume.

File: db/resources/bin/pg_backup.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+###########################
+
+# Make sure we're running as the required backup user
+if [ "$BACKUP_USER" != "" -a "$(id -un)" != "$BACKUP_USER" ] ; then
+	echo "This script must be run as $BACKUP_USER. Exiting." 1>&2
+	exit 1
+fi
+
+###########################
+### INITIALISE DEFAULTS ###
```
[the remark this answers is not in the document]
Comment (jdk24) [reply db/resources/bin/pg_backup.sh:43]: Oi 🇬🇧

File: db/resources/bin/pg_backup.sh [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+	
+	exit 0;
+fi
+
+# DAILY BACKUPS
+
+# Delete daily backups 7 days old or more
+find $BACKUP_DIR -maxdepth 1 -mtime +$DAYS_TO_KEEP -name "*-daily" -exec rm -rf '{}' ';'
+
+perform_backups "-daily"
```
[the remark this answers is not in the document]
Comment (jdk24) [reply db/resources/bin/pg_backup.sh]: Added in new commit. Thank you!

File: db/resources/bin/pg_backup.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrismayer) [reply db/resources/bin/pg_backup.sh]: oh, right

**Comment by chrismayer** [comment]
@jdk24 service log while starting says at the end `psql:/update-files.d/012_alter_lubw_parser.sql:161: ERROR:  schema "station_data" does not exist`

**Comment by chrismayer** [comment]
With the patch [here](https://github.com/meggsimum/sauber-sdi-docker/pull/132/files#diff-c272aa557400cbf25004c8035bd2d4aaddd4286ae997a072bd8d54f1a5009005R1) it works.

image

**Review by chrismayer — APPROVED** [review]

**Comment by jdk24** [comment]
Thank you for finding this!! Merging.

**Pull request merged**